lib/xero_gateway/gateway.rb

```diff
@@ -791,7 +808,7 @@ def parse_response(raw_response, request = {}, options = {})
       end if response_element
 
       # If a single result is returned don't put it in an array
-      if response.response_item.is_a?(Array) && response.response_item.size == 1
+      if response.response_item.is_a?(Array) && response.response_item.size == 1 && !parse_options[:dont_flatten]
```

Comment:
This is a good solution for now, but we should really build up a list of the entities that are "flattenable" and those that just happen to be returning a single value :)